packages/block-editor/src/components/list-view/index.js

```diff
@@ -168,6 +173,26 @@ function ListView(
 		]
 	);
 
+	// If a selection is made outside the block list,
+	// for example, in the Block Editor,
+	// try to expand the block list tree.
+	useEffect( () => {
+		if (
+			! hasFocus &&
+			Array.isArray( selectedBlockParentClientIds ) &&
+			selectedBlockParentClientIds.length
+		) {
+			selectedBlockParentClientIds.forEach( ( clientId ) => {
+				if ( ! expandedState[ clientId ] ) {
+					setExpandedState( {
+						type: 'expand',
+						clientId,
+					} );
```

Comment:
Two recommendations here:
- Let's make a new action so we can dispatch a single one instead of many
- It might be worth exploring if we need to keep track of focus or not. (Is there any other state we can query?) I'd timebox this.

Reply:
> Let's make a new action so we can dispatch a single one instead of many

Good idea. I'll make a start next week. Thanks again, and great work.